packages/dashboard-frontend/src/store/GitOauthConfig/index.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { Action, Reducer } from 'redux';
+import common, { api } from '@eclipse-che/common';
+import { AppThunk } from '..';
+import { createObject } from '../helpers';
+import { AUTHORIZED } from '../sanityCheckMiddleware';
+import { container } from '../../inversify.config';
+import { CheWorkspaceClient } from '../../services/workspace-client/cheworkspace/cheWorkspaceClient';
+
+export interface State {
+  isLoading: boolean;
+  gitOauth: { name: api.GitOauthProvider; endpointUrl: string }[];
```

Comment:
Why don't you extract `{ name: api.GitOauthProvider; endpointUrl: string }` as an interface or type?

Reply:
I have fixed it.